addon/components/sl-menu.js

```diff
@@ -232,415 +94,555 @@ export default Ember.Component.extend({
-    keyHandler: false,
+    // -------------------------------------------------------------------------
+    // Observers
 
     /**
-     * Is the menu being interacted with via the keyboard?
+     * Initialize any computed properties that need setup
      *
-     * @type {Boolean}
+     * @function
+     * @returns {undefined}
      */
-    keyboardInUse: false,
+    initialize: Ember.on(
+        'init',
+        function() {
+            this.set( 'selections', new Ember.A() );
+        }
+    ),
 
     /**
-     * Whether "Show All" icon and functionality is enabled
-     *
-     * This is a string representaton of a boolean state.
+     * Setup the stream actions bindings
      *
-     * @type {String}
+     * @function
+     * @returns {undefined}
      */
-    showAll: 'false',
+    setupStreamActions: Ember.on(
+        'init',
+        function() {
+            const stream = this.get( 'stream' );
+
+            if ( stream ) {
+                stream.on( 'hideAll', () => {
+                    this.hideAll();
+                });
+
+                stream.on( 'showAll', () => {
+                    this.showAll();
+                });
+
+                stream.on( 'select', ( indexOrDirection ) => {
+                    if ( 'number' === Ember.typeOf( indexOrDirection ) ) {
+                        this.select( indexOrDirection );
+                        return;
+                    }
+
+                    switch ( indexOrDirection ) {
+                        case Direction.DOWN:
+                            this.selectDown();
+                            break;
+
+                        case Direction.LEFT:
+                            this.selectLeft();
+                            break;
+
+                        case Direction.NEXT:
+                            this.selectNext();
+                            break;
+
+                        case Direction.PARENT:
+                            this.selectParent();
+                            break;
+
+                        case Direction.PREVIOUS:
+                            this.selectPrevious();
+                            break;
+
+                        case Direction.RIGHT:
+                            this.selectRight();
+                            break;
+
+                        case Direction.SUB_MENU:
+                            this.selectSubMenu();
+                            break;
+
+                        case Direction.UP:
+                            this.selectUp();
+                            break;
+
+                        default:
+                            warn( `Received "select" stream action with invalid selection, "${indexOrDirection}"` );
+                            break;
+                    }
+                });
+            }
+        }
+    ),
 
     /**
-     * When true, allows key binding to drill down
+     * Retrieve the currently selected item
      *
-     * @type {Boolean}
+     * @function
+     * @returns {?Object}
```

Comment:
Same question as previous comment about this saying can return `null` but no code to support this, and whether `null` is supposed to be the default return value of computed properties.